### PR TITLE
[Mailer] Logger vs debug mailer

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Currently, there is no way to get the network data for the HTTP calls done by the HTTP transports (which makes debugging harder). For SMTP, we do have the network data, but as logs (each SMTP command/response is its own log line which means that the logs are "polluted" and the data is not tied with the sent message).

This pull request adds a `getDebug()` method on `SentMessage`. That allows to get the debug data conveniently in a standardized way (for both SMTP and HTTP transports). I have moved the SMTP logs to this new mechanism and added support for HTTP transports.

### CODE REVIEW

**Comment by fabpot** [comment]
Failures on high deps will be resolved when merging up.
Failures on low are not related.

File: src/Symfony/Component/Mailer/Exception/HttpTransportException.php [hunk 1]
```diff
@@ -11,9 +11,24 @@
...
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
 /**
  * @author Fabien Potencier <fabien@symfony.com>
  */
 class HttpTransportException extends TransportException
 {
+    private $response;
+
+    public function __construct(string $message = null, ResponseInterface $response, int $code = 0, \Exception $previous = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Mailer/Exception/HttpTransportException.php]: do we ever use $previous? should we remove it if not?

Comment (fabpot) [reply src/Symfony/Component/Mailer/Exception/HttpTransportException.php]: They allow future usage. I'd like to keep them.

File: src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php [hunk 2]
```diff
@@ -28,8 +28,14 @@ abstract class AbstractStream
     protected $in;
     protected $out;
 
+    private $debug = '';
+
     public function write(string $bytes): void
     {
+        foreach (explode("\n", trim($bytes)) as $line) {
+            $this->debug .= sprintf("> %s\n", $line);
```
[the remark this answers is not in the document]
Comment (fabpot) [reply src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php]: Done, the body is excluded now

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**